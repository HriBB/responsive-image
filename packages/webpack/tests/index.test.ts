import compiler from './compiler';

test('it produces expected output', async () => {
  const stats = (await compiler('fixtures/image.jpg?responsive', {})).toJson({
    source: true,
  });

  expect(stats.modules).toBeDefined();
  expect(stats.modules![0]?.modules).toHaveLength(2);

  const output = stats.modules?.[0]?.modules?.[0]?.source;
  expect(output).toMatchSnapshot();
});

test('custom loader options are supported', async () => {
  const stats = (
    await compiler('fixtures/image.jpg?responsive', {
      widths: [1000, 2000],
      formats: ['original', 'avif'],
      name: 'test-[width].[ext]',
      webPath: 'https://cdn.example.com/images',
    })
  ).toJson({
    source: true,
  });

  expect(stats.modules).toBeDefined();
  expect(stats.modules![0]?.modules).toHaveLength(2);

  const output = stats.modules?.[0]?.modules?.[0]?.source;
  expect(output).toMatchSnapshot();
});

test('custom query params are supported', async () => {
  const stats = (
    await compiler(
      'fixtures/image.jpg?lqip=color&widths=100,200&formats=webp&responsive',
      {},
    )
  ).toJson({
    source: true,
  });

  expect(stats.modules).toBeDefined();
  expect(stats.modules![0]?.modules).toHaveLength(3);

  const output = stats.modules?.[0]?.modules?.[0]?.source;
  expect(output).toMatchSnapshot();
});

describe('LQIP', function () {
  test('color LQIP is supported', async () => {
    const stats = (
      await compiler('fixtures/image.jpg?responsive', {
        lqip: { type: 'color' },
      })
    ).toJson({
      source: true,
    });

    expect(stats.modules).toBeDefined();
    expect(stats.modules![0]?.modules).toHaveLength(3);

    const output = stats.modules?.[0]?.modules?.[0]?.source;
    expect(output).toMatchSnapshot();
  });

  test('inline LQIP is supported', async () => {
    const stats = (
      await compiler('fixtures/image.jpg?responsive', {
        lqip: { type: 'inline' },
      })
    ).toJson({
      source: true,
    });

    expect(stats.modules).toBeDefined();
    expect(stats.modules![0]?.modules).toHaveLength(3);

    const output = stats.modules?.[0]?.modules?.[0]?.source;
    expect(output).toMatchSnapshot();
  });

  test('blurhash LQIP is supported', async () => {
    const stats = (
      await compiler('fixtures/image.jpg?responsive', {
        lqip: { type: 'blurhash' },
      })
    ).toJson({
      source: true,
    });

    expect(stats.modules).toBeDefined();
    expect(stats.modules![0]?.modules).toHaveLength(2);

    const output = stats.modules?.[0]?.modules?.[0]?.source;
    expect(output).toMatchSnapshot();
  });
});
