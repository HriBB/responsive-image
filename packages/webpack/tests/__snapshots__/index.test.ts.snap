// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LQIP > blurhash LQIP is supported 1`] = `
"import { findMatchingImage } from '@responsive-image/core';
const images = [{"url":__webpack_public_path__ + "images/image-640w.jpg","width":640,"format":"jpeg"},{"url":__webpack_public_path__ + "images/image-640w.webp","width":640,"format":"webp"},{"url":__webpack_public_path__ + "images/image-750w.jpg","width":750,"format":"jpeg"},{"url":__webpack_public_path__ + "images/image-750w.webp","width":750,"format":"webp"},{"url":__webpack_public_path__ + "images/image-828w.jpg","width":828,"format":"jpeg"},{"url":__webpack_public_path__ + "images/image-828w.webp","width":828,"format":"webp"},{"url":__webpack_public_path__ + "images/image-1080w.jpg","width":1080,"format":"jpeg"},{"url":__webpack_public_path__ + "images/image-1080w.webp","width":1080,"format":"webp"},{"url":__webpack_public_path__ + "images/image-1200w.jpg","width":1200,"format":"jpeg"},{"url":__webpack_public_path__ + "images/image-1200w.webp","width":1200,"format":"webp"},{"url":__webpack_public_path__ + "images/image-1920w.jpg","width":1920,"format":"jpeg"},{"url":__webpack_public_path__ + "images/image-1920w.webp","width":1920,"format":"webp"},{"url":__webpack_public_path__ + "images/image-2048w.jpg","width":2048,"format":"jpeg"},{"url":__webpack_public_path__ + "images/image-2048w.webp","width":2048,"format":"webp"},{"url":__webpack_public_path__ + "images/image-3840w.jpg","width":3840,"format":"jpeg"},{"url":__webpack_public_path__ + "images/image-3840w.webp","width":3840,"format":"webp"}];
export default {
    imageTypes: ["jpeg","webp"],
    availableWidths: [640,750,828,1080,1200,1920,2048,3840],
    lqip: {"type":"blurhash","hash":"NYGkgjIp9b%1f6Iq~UaeV@WYWVs,%MoJWBWCjZbH","width":6,"height":3},
    aspectRatio: 2,
    imageUrlFor(w, f) {
      return findMatchingImage(images, w, f ?? "jpeg")?.url;
    }
  }"
`;

exports[`LQIP > color LQIP is supported 1`] = `
"import { findMatchingImage } from '@responsive-image/core';
import '/fixtures/image.jpg.css!=!@responsive-image/webpack/lqip/color-css!/fixtures/image.jpg?className=eri-dyn-1';
const images = [{"url":__webpack_public_path__ + "images/image-640w.jpg","width":640,"format":"jpeg"},{"url":__webpack_public_path__ + "images/image-640w.webp","width":640,"format":"webp"},{"url":__webpack_public_path__ + "images/image-750w.jpg","width":750,"format":"jpeg"},{"url":__webpack_public_path__ + "images/image-750w.webp","width":750,"format":"webp"},{"url":__webpack_public_path__ + "images/image-828w.jpg","width":828,"format":"jpeg"},{"url":__webpack_public_path__ + "images/image-828w.webp","width":828,"format":"webp"},{"url":__webpack_public_path__ + "images/image-1080w.jpg","width":1080,"format":"jpeg"},{"url":__webpack_public_path__ + "images/image-1080w.webp","width":1080,"format":"webp"},{"url":__webpack_public_path__ + "images/image-1200w.jpg","width":1200,"format":"jpeg"},{"url":__webpack_public_path__ + "images/image-1200w.webp","width":1200,"format":"webp"},{"url":__webpack_public_path__ + "images/image-1920w.jpg","width":1920,"format":"jpeg"},{"url":__webpack_public_path__ + "images/image-1920w.webp","width":1920,"format":"webp"},{"url":__webpack_public_path__ + "images/image-2048w.jpg","width":2048,"format":"jpeg"},{"url":__webpack_public_path__ + "images/image-2048w.webp","width":2048,"format":"webp"},{"url":__webpack_public_path__ + "images/image-3840w.jpg","width":3840,"format":"jpeg"},{"url":__webpack_public_path__ + "images/image-3840w.webp","width":3840,"format":"webp"}];
export default {
    imageTypes: ["jpeg","webp"],
    availableWidths: [640,750,828,1080,1200,1920,2048,3840],
    lqip: {"type":"color","class":"eri-dyn-1"},
    aspectRatio: 2,
    imageUrlFor(w, f) {
      return findMatchingImage(images, w, f ?? "jpeg")?.url;
    }
  }"
`;

exports[`LQIP > inline LQIP is supported 1`] = `
"import { findMatchingImage } from '@responsive-image/core';
import '/fixtures/image.jpg.css!=!@responsive-image/webpack/lqip/inline-css!/fixtures/image.jpg?className=eri-dyn-1&targetPixels=60';
const images = [{"url":__webpack_public_path__ + "images/image-640w.jpg","width":640,"format":"jpeg"},{"url":__webpack_public_path__ + "images/image-640w.webp","width":640,"format":"webp"},{"url":__webpack_public_path__ + "images/image-750w.jpg","width":750,"format":"jpeg"},{"url":__webpack_public_path__ + "images/image-750w.webp","width":750,"format":"webp"},{"url":__webpack_public_path__ + "images/image-828w.jpg","width":828,"format":"jpeg"},{"url":__webpack_public_path__ + "images/image-828w.webp","width":828,"format":"webp"},{"url":__webpack_public_path__ + "images/image-1080w.jpg","width":1080,"format":"jpeg"},{"url":__webpack_public_path__ + "images/image-1080w.webp","width":1080,"format":"webp"},{"url":__webpack_public_path__ + "images/image-1200w.jpg","width":1200,"format":"jpeg"},{"url":__webpack_public_path__ + "images/image-1200w.webp","width":1200,"format":"webp"},{"url":__webpack_public_path__ + "images/image-1920w.jpg","width":1920,"format":"jpeg"},{"url":__webpack_public_path__ + "images/image-1920w.webp","width":1920,"format":"webp"},{"url":__webpack_public_path__ + "images/image-2048w.jpg","width":2048,"format":"jpeg"},{"url":__webpack_public_path__ + "images/image-2048w.webp","width":2048,"format":"webp"},{"url":__webpack_public_path__ + "images/image-3840w.jpg","width":3840,"format":"jpeg"},{"url":__webpack_public_path__ + "images/image-3840w.webp","width":3840,"format":"webp"}];
export default {
    imageTypes: ["jpeg","webp"],
    availableWidths: [640,750,828,1080,1200,1920,2048,3840],
    lqip: {"type":"inline","class":"eri-dyn-1"},
    aspectRatio: 2,
    imageUrlFor(w, f) {
      return findMatchingImage(images, w, f ?? "jpeg")?.url;
    }
  }"
`;

exports[`custom loader options are supported 1`] = `
"import { findMatchingImage } from '@responsive-image/core';
const images = [{"url":"https:/cdn.example.com/images/test-1000.jpg","width":1000,"format":"jpeg"},{"url":"https:/cdn.example.com/images/test-1000.avif","width":1000,"format":"avif"},{"url":"https:/cdn.example.com/images/test-2000.jpg","width":2000,"format":"jpeg"},{"url":"https:/cdn.example.com/images/test-2000.avif","width":2000,"format":"avif"}];
export default {
    imageTypes: ["jpeg","avif"],
    availableWidths: [1000,2000],
    
    aspectRatio: 2,
    imageUrlFor(w, f) {
      return findMatchingImage(images, w, f ?? "jpeg")?.url;
    }
  }"
`;

exports[`custom query params are supported 1`] = `
"import { findMatchingImage } from '@responsive-image/core';
import '/fixtures/image.jpg.css!=!@responsive-image/webpack/lqip/color-css!/fixtures/image.jpg?className=eri-dyn-0';
const images = [{"url":__webpack_public_path__ + "images/image-100w.webp","width":100,"format":"webp"},{"url":__webpack_public_path__ + "images/image-200w.webp","width":200,"format":"webp"}];
export default {
    imageTypes: ["webp"],
    availableWidths: [100,200],
    lqip: {"type":"color","class":"eri-dyn-0"},
    aspectRatio: 2,
    imageUrlFor(w, f) {
      return findMatchingImage(images, w, f ?? "webp")?.url;
    }
  }"
`;

exports[`it produces expected output 1`] = `
"import { findMatchingImage } from '@responsive-image/core';
const images = [{"url":__webpack_public_path__ + "images/image-640w.jpg","width":640,"format":"jpeg"},{"url":__webpack_public_path__ + "images/image-640w.webp","width":640,"format":"webp"},{"url":__webpack_public_path__ + "images/image-750w.jpg","width":750,"format":"jpeg"},{"url":__webpack_public_path__ + "images/image-750w.webp","width":750,"format":"webp"},{"url":__webpack_public_path__ + "images/image-828w.jpg","width":828,"format":"jpeg"},{"url":__webpack_public_path__ + "images/image-828w.webp","width":828,"format":"webp"},{"url":__webpack_public_path__ + "images/image-1080w.jpg","width":1080,"format":"jpeg"},{"url":__webpack_public_path__ + "images/image-1080w.webp","width":1080,"format":"webp"},{"url":__webpack_public_path__ + "images/image-1200w.jpg","width":1200,"format":"jpeg"},{"url":__webpack_public_path__ + "images/image-1200w.webp","width":1200,"format":"webp"},{"url":__webpack_public_path__ + "images/image-1920w.jpg","width":1920,"format":"jpeg"},{"url":__webpack_public_path__ + "images/image-1920w.webp","width":1920,"format":"webp"},{"url":__webpack_public_path__ + "images/image-2048w.jpg","width":2048,"format":"jpeg"},{"url":__webpack_public_path__ + "images/image-2048w.webp","width":2048,"format":"webp"},{"url":__webpack_public_path__ + "images/image-3840w.jpg","width":3840,"format":"jpeg"},{"url":__webpack_public_path__ + "images/image-3840w.webp","width":3840,"format":"webp"}];
export default {
    imageTypes: ["jpeg","webp"],
    availableWidths: [640,750,828,1080,1200,1920,2048,3840],
    
    aspectRatio: 2,
    imageUrlFor(w, f) {
      return findMatchingImage(images, w, f ?? "jpeg")?.url;
    }
  }"
`;
